tests/functional_simulator/functional_simulator_integration_tests.cpp

```diff
@@ -1162,3 +1162,332 @@ TEST_F(IntegrationTest, rawDependencyChaining) {
     EXPECT_EQ(stats.getCategoryCount(mips_lite::InstructionCategory::CONTROL_FLOW), 1);
     EXPECT_EQ(stats.getCategoryCount(mips_lite::InstructionCategory::MEMORY_ACCESS), 0);
 }
+
+// AND logical testing
+TEST_F(IntegrationTest, AND_NoForwarding) {
+    std::vector<uint32_t> program = {
+        0x0402ff00,  // ADDI R2 R0 -256
+        0x0403c3c9,  // ADDI R3 R0 -15415
+        0x0404ffff,  // ADDI R4 R0 -1
+        0x20437000,  // AND R14 R2 R3
+        0x20407800,  // AND R15 R2 R0
+        0x20448000,  // AND R16 R2 R4
+        0x247100ff,  // ANDI R17 R3 255
+        0x24720000,  // ANDI R18 R3 0
+        0x24730001,  // ANDI R19 R3 1
+        0x44000000   // HALT
+    };
+
+    setupMockMemory(mem, program);
+
+    while (!sim_no_forward->isProgramFinished()) {
+        sim_no_forward->cycle();
+        if (stats.getClockCycles() >= 1000) {
+            ADD_FAILURE() << "Simulator did not halt within 1000 cycles";
+            break;
+        }
+    }
+
+    EXPECT_EQ(rf.read(14), 0xFFFFC300);
+    EXPECT_EQ(rf.read(15), 0x00000000);
+    EXPECT_EQ(rf.read(16), 0xFFFFFF00);
+    EXPECT_EQ(rf.read(17), 0x000000C9);
+    EXPECT_EQ(rf.read(18), 0x00000000);
+    EXPECT_EQ(rf.read(19), 0x00000001);
+    EXPECT_EQ(stats.totalInstructions(), 10);
+
+    EXPECT_EQ(sim_no_forward->getPC(), 40);  // 10 instructions * 4
+    EXPECT_EQ(stats.getStalls(), 1);         // simulator reports 1
+    EXPECT_EQ(stats.getClockCycles(), 15);   // as measured
+    EXPECT_EQ(stats.getCategoryCount(mips_lite::InstructionCategory::ARITHMETIC), 3);
+    EXPECT_EQ(stats.getCategoryCount(mips_lite::InstructionCategory::LOGICAL), 6);
+    EXPECT_EQ(stats.getCategoryCount(mips_lite::InstructionCategory::CONTROL_FLOW), 1);
+    EXPECT_EQ(stats.getCategoryCount(mips_lite::InstructionCategory::MEMORY_ACCESS), 0);
+}
+
+TEST_F(IntegrationTest, AND_WithForwarding) {
+    std::vector<uint32_t> program = {
+        0x0402ff00,  // ADDI R2 R0 -256
+        0x0403c3c9,  // ADDI R3 R0 -15415
+        0x0404ffff,  // ADDI R4 R0 -1
+        0x20437000,  // AND R14 R2 R3
+        0x20407800,  // AND R15 R2 R0
+        0x20448000,  // AND R16 R2 R4
+        0x247100ff,  // ANDI R17 R3 255
+        0x24720000,  // ANDI R18 R3 0
+        0x24730001,  // ANDI R19 R3 1
+        0x44000000   // HALT
+    };
+
+    resetSimulator(sim_with_forward, rf, stats, mem, true);
```

Comment:
This doesn't hurt anything, but also shouldn't be necessary. This reset is used when the forward and no-forward simulators are tested within the same `TEST_F` block. Since they share the `stats` and `rf`, those need to be reset. Otherwise, if the tests are in different `TEST_F` blocks, the `setUp` function should do the reset work we need.